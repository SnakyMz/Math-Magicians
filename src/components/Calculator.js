import React, { useState } from 'react';
import calculate from '../logic/calculator';

function Calculator() {
  const [state, setState] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleClick = (btnName) => {
    const result = calculate(state, btnName);
    setState(result);
  };

  return (
    <section id="calculator">
      <h2>Let&apos;s do some math</h2>
      <div className="operators">
        <span className="display">
          {`${state.total || ''}${state.operation || ''}${state.next || ''}` || '0'}
        </span>
        <button type="button" className="grey" onClick={() => handleClick('AC')}>AC</button>
        <button type="button" className="grey" onClick={() => handleClick('+/-')}>+/-</button>
        <button type="button" className="grey" onClick={() => handleClick('%')}>%</button>
        <button type="button" className="orange" onClick={() => handleClick('รท')}>รท</button>
        <button type="button" className="grey" onClick={() => handleClick('7')}>7</button>
        <button type="button" className="grey" onClick={() => handleClick('8')}>8</button>
        <button type="button" className="grey" onClick={() => handleClick('9')}>9</button>
        <button type="button" className="orange" onClick={() => handleClick('x')}>x</button>
        <button type="button" className="grey" onClick={() => handleClick('4')}>4</button>
        <button type="button" className="grey" onClick={() => handleClick('5')}>5</button>
        <button type="button" className="grey" onClick={() => handleClick('6')}>6</button>
        <button type="button" className="orange" onClick={() => handleClick('-')}>-</button>
        <button type="button" className="grey" onClick={() => handleClick('1')}>1</button>
        <button type="button" className="grey" onClick={() => handleClick('2')}>2</button>
        <button type="button" className="grey" onClick={() => handleClick('3')}>3</button>
        <button type="button" className="orange" onClick={() => handleClick('+')}>+</button>
        <button type="button" className="grey zero" onClick={() => handleClick('0')}>0</button>
        <button type="button" className="grey" onClick={() => handleClick('.')}>.</button>
        <button type="button" className="orange" onClick={() => handleClick('=')}>=</button>
      </div>
    </section>
  );
}

export default Calculator;
