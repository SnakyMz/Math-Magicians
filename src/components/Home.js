function Home() {
  return (
    <section id="home">
      <h2>Welcome to our page</h2>
      <p>
        This webpage contains a calculator that perfomrs basic calculations.
        You can perform addition, subtraction, multiplication, division, and percentage operations.
        The &quot;AC&quot; button resets the calculations to zero.
        This webpage also contains a quote section.
        The Quote section generates a random quote on load.
      </p>
    </section>
  );
}

export default Home;
