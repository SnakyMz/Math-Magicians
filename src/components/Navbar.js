import { Link } from 'react-router-dom';

function Navbar() {
  return (
    <header>
      <h1>Math Magicians</h1>
      <nav>
        <ul>
          <li><Link to="/" className="navitem">Home</Link></li>
          <span>|</span>
          <li><Link to="/Calculator" className="navitem">Calculator</Link></li>
          <span>|</span>
          <li><Link to="/Quotes" className="navitem">Quotes</Link></li>
        </ul>
      </nav>
    </header>
  );
}

export default Navbar;
