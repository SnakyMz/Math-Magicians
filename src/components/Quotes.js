import { useEffect, useState } from 'react';

function Quotes() {
  const [data, setData] = useState('');
  const [isLoading, setIsLoading] = useState('');
  const [hasError, setHasError] = useState('');

  useEffect(() => {
    const fetchData = async () => {
      setIsLoading(true);
      try {
        const responce = await fetch('https://api.api-ninjas.com/v1/quotes', {
          method: 'GET',
          headers: {
            'X-Api-Key': 'JKZPHBt7RyLSIBui4mVJSQ==gv0jhhogQan4jcrl',
          },
        });
        const json = await responce.json();
        setData({
          quote: json[0].quote,
          author: json[0].author,
        });
      } catch (error) {
        setHasError(true);
      }
      setIsLoading(false);
    };
    fetchData();
  }, [setData, setIsLoading]);

  if (isLoading) return <div id="loading">Loading...</div>;
  if (hasError) return <div id="error">Something went wrong!</div>;

  return (
    <span className="quote">
      &quot;
      {data.quote}
      &quot;
      -
      &nbsp;
      {data.author}
    </span>
  );
}

export default Quotes;
