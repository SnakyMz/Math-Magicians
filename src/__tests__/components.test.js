import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import Navbar from '../components/Navbar';
import Home from '../components/Home';
import Calculator from '../components/Calculator';
import Quotes from '../components/Quotes';

describe('Testing Rendering', () => {
  it('Should render Navbar', () => {
    const nav = render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );
    expect(nav).toMatchSnapshot();
  });

  it('Should render Home page', () => {
    const home = render(<Home />);
    expect(home).toMatchSnapshot();
  });

  it('Should render Calculator page', () => {
    const calcy = render(<Calculator />);
    expect(calcy).toMatchSnapshot();
  });

  it('Should render Quotes page', () => {
    const quot = render(<Quotes />);
    expect(quot).toMatchSnapshot();
  });
});
