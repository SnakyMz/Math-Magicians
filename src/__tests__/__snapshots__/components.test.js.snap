// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Rendering Should render Calculator page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        id="calculator"
      >
        <h2>
          Let's do some math
        </h2>
        <div
          class="operators"
        >
          <span
            class="display"
          >
            0
          </span>
          <button
            class="grey"
            type="button"
          >
            AC
          </button>
          <button
            class="grey"
            type="button"
          >
            +/-
          </button>
          <button
            class="grey"
            type="button"
          >
            %
          </button>
          <button
            class="orange"
            type="button"
          >
            รท
          </button>
          <button
            class="grey"
            type="button"
          >
            7
          </button>
          <button
            class="grey"
            type="button"
          >
            8
          </button>
          <button
            class="grey"
            type="button"
          >
            9
          </button>
          <button
            class="orange"
            type="button"
          >
            x
          </button>
          <button
            class="grey"
            type="button"
          >
            4
          </button>
          <button
            class="grey"
            type="button"
          >
            5
          </button>
          <button
            class="grey"
            type="button"
          >
            6
          </button>
          <button
            class="orange"
            type="button"
          >
            -
          </button>
          <button
            class="grey"
            type="button"
          >
            1
          </button>
          <button
            class="grey"
            type="button"
          >
            2
          </button>
          <button
            class="grey"
            type="button"
          >
            3
          </button>
          <button
            class="orange"
            type="button"
          >
            +
          </button>
          <button
            class="grey zero"
            type="button"
          >
            0
          </button>
          <button
            class="grey"
            type="button"
          >
            .
          </button>
          <button
            class="orange"
            type="button"
          >
            =
          </button>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      id="calculator"
    >
      <h2>
        Let's do some math
      </h2>
      <div
        class="operators"
      >
        <span
          class="display"
        >
          0
        </span>
        <button
          class="grey"
          type="button"
        >
          AC
        </button>
        <button
          class="grey"
          type="button"
        >
          +/-
        </button>
        <button
          class="grey"
          type="button"
        >
          %
        </button>
        <button
          class="orange"
          type="button"
        >
          รท
        </button>
        <button
          class="grey"
          type="button"
        >
          7
        </button>
        <button
          class="grey"
          type="button"
        >
          8
        </button>
        <button
          class="grey"
          type="button"
        >
          9
        </button>
        <button
          class="orange"
          type="button"
        >
          x
        </button>
        <button
          class="grey"
          type="button"
        >
          4
        </button>
        <button
          class="grey"
          type="button"
        >
          5
        </button>
        <button
          class="grey"
          type="button"
        >
          6
        </button>
        <button
          class="orange"
          type="button"
        >
          -
        </button>
        <button
          class="grey"
          type="button"
        >
          1
        </button>
        <button
          class="grey"
          type="button"
        >
          2
        </button>
        <button
          class="grey"
          type="button"
        >
          3
        </button>
        <button
          class="orange"
          type="button"
        >
          +
        </button>
        <button
          class="grey zero"
          type="button"
        >
          0
        </button>
        <button
          class="grey"
          type="button"
        >
          .
        </button>
        <button
          class="orange"
          type="button"
        >
          =
        </button>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing Rendering Should render Home page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        id="home"
      >
        <h2>
          Welcome to our page
        </h2>
        <p>
          This webpage contains a calculator that perfomrs basic calculations.
          <br />
          You can perform addition, subtraction, multiplication, division, and percentage operations.
          <br />
          The "AC" button resets the calculations to zero.
          <br />
          This webpage also contains a quote section.
          <br />
          The Quote section generates a random quote on load.
        </p>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      id="home"
    >
      <h2>
        Welcome to our page
      </h2>
      <p>
        This webpage contains a calculator that perfomrs basic calculations.
        <br />
        You can perform addition, subtraction, multiplication, division, and percentage operations.
        <br />
        The "AC" button resets the calculations to zero.
        <br />
        This webpage also contains a quote section.
        <br />
        The Quote section generates a random quote on load.
      </p>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing Rendering Should render Navbar 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <h1>
          Math Magicians
        </h1>
        <nav>
          <ul>
            <li>
              <a
                class="navitem"
                href="/"
              >
                Home
              </a>
            </li>
            <span>
              |
            </span>
            <li>
              <a
                class="navitem"
                href="/Calculator"
              >
                Calculator
              </a>
            </li>
            <span>
              |
            </span>
            <li>
              <a
                class="navitem"
                href="/Quotes"
              >
                Quotes
              </a>
            </li>
          </ul>
        </nav>
      </header>
    </div>
  </body>,
  "container": <div>
    <header>
      <h1>
        Math Magicians
      </h1>
      <nav>
        <ul>
          <li>
            <a
              class="navitem"
              href="/"
            >
              Home
            </a>
          </li>
          <span>
            |
          </span>
          <li>
            <a
              class="navitem"
              href="/Calculator"
            >
              Calculator
            </a>
          </li>
          <span>
            |
          </span>
          <li>
            <a
              class="navitem"
              href="/Quotes"
            >
              Quotes
            </a>
          </li>
        </ul>
      </nav>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing Rendering Should render Quotes page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="loading"
      >
        Loading...
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      id="loading"
    >
      Loading...
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
